src/DocumentSession.cpp

```diff
@@ -3,3 +3,63 @@
 // found in the LICENSE file.
 
 #include "DocumentSession.h"
+
+DocumentSession::DocumentSession()
+{
+    doc_manager_ = new DocumentManager();
+}
+
+void DocumentSession::addSession(SocketSession *session, const int id)
+{
+    sessions_.insert(session);
+    std::vector<Revision> *revisions_ = doc_manager_->getHistoryFromDocument(id);
+    if (revisions_ != nullptr)
+    {
+        for (const auto &revision : *revisions_)
+        {
+            session->sendMessage(SocketUtility::parseMessage(revision.toString()));
+        }
+    }
+}
+
+void DocumentSession::removeSession(SocketSession *session)
+{
+    sessions_.erase(session);
+}
+
+void DocumentSession::parseUpdateParameters(const std::string &update_params)
+{
+    std::string update_params_ = update_params;
+    std::string delimiter = ",";
+
+    size_t pos = 0;
+    std::string token;
+    std::vector<std::string> param_list;
+    while ((pos = update_params_.find(delimiter)) != std::string::npos)
```

Comment:
[Optional] This operation is known as "tokenize". There's many library to do this, including boost tokenizer and strtok().